tests/test_meshtags.py

```diff
@@ -0,0 +1,284 @@
+    """
+    Helper function to generate map from meshtag value to its corresponding index and midpoint.
+
+    Args:
+        mesh: The mesh
+        meshtag: The associated meshtag
+        comm: MPI communicator to gather the map from all processes with
+        root (int): Rank to store data on
+    Returns:
+        Root rank returns the map, all other ranks return None
+    """
+    midpoints = dolfinx.mesh.compute_midpoints(mesh, meshtag.dim, meshtag.indices)
+    e_map = mesh.topology.index_map(meshtag.dim)
+    value_to_midpoint = {}
+    for index, value in zip(meshtag.indices, meshtag.values):
+        value_to_midpoint[value] = (
+            int(e_map.local_range[0] + index),
+            midpoints[index],
+        )
+    global_map = comm.gather(value_to_midpoint, root=root)
+    if comm.rank == root:
+        return dict(ChainMap(*global_map))  # type: ignore
+    return None
+
+
+@pytest.mark.parametrize("read_mode", read_modes)
+@pytest.mark.parametrize("read_comm", [MPI.COMM_SELF, MPI.COMM_WORLD])
+def test_checkpointing_meshtags_1D(mesh_1D, read_comm, read_mode):
+    mesh = mesh_1D
+
+    # Write unique mesh file for each combination of MPI communicator and dtype
+    hash = f"{mesh.comm.size}_{mesh.geometry.x.dtype}"
+    filename = f"meshtags_1D_{hash}.bp"
+
+    # If mesh communicator is more than a self communicator or serial write on all processes.
+    # If serial or self communicator, only write on root rank
+    if mesh.comm.size != 1:
+        adios4dolfinx.write_mesh(mesh, filename, engine="BP4")
+    else:
+        if MPI.COMM_WORLD.rank == root:
+            adios4dolfinx.write_mesh(mesh, filename, engine="BP4")
+
+    # Create meshtags labeling each entity (of each co-dimension) with a
+    # unique number (their initial global index).
+    org_maps = []
+    for dim in range(mesh.topology.dim + 1):
+
+        mesh.topology.create_connectivity(dim, mesh.topology.dim)
+        e_map = mesh.topology.index_map(dim)
+        num_entities_local = e_map.size_local
+        entities = np.arange(num_entities_local, dtype=np.int32)
+        ft = dolfinx.mesh.meshtags(mesh, dim, entities, e_map.local_range[0] + entities)
+        ft.name = f"entity_{dim}"
+
+        # If parallel write on all processes, else write on root rank
+        if mesh.comm.size != 1:
+            adios4dolfinx.write_meshtags(filename, mesh, ft, engine="BP4")
+            # Create map from mesh tag value to its corresponding index and midpoint
+            org_map = generate_reference_map(mesh, ft, mesh.comm, root)
+            org_maps.append(org_map)
+        else:
+            if MPI.COMM_WORLD.rank == root:
+                adios4dolfinx.write_meshtags(filename, mesh, ft, engine="BP4")
+                # Create map from mesh tag value to its corresponding index and midpoint
+                org_map = generate_reference_map(mesh, ft, MPI.COMM_SELF, root)
+                org_maps.append(org_map)
+        del ft
+    del mesh
```

Comment:
Why do we need to delete `ft` and `mesh` here?

Reply:
Just want to make sure we dont use then later on in verification:)